drivers/usb_c/tcpc/ucpd_stm32.c

```diff
@@ -1135,7 +1135,7 @@ static void ucpd_isr(const struct device *dev_inst[])
 	 */
 	if (sr & tx_done_mask) {
 		/* Check for tx message complete */
-		if (sr & (UCPD_SR_TXMSGSENT | UCPD_SR_HRSTSENT)) {
+		if (sr & UCPD_SR_TXMSGSENT) {
```

Comment:
This comment might be orthogonal to this PR, but it seems like this code is treating this register as if only one of these bits at a time will be set. Is that how it works?

Reply:
Multiple bits can be set and I've been thinking about restructuring this code. Maybe test the HR bits first, followed by the other.

Reply:
The way it's set up right now, if multiple bits are set at the same time, does the code react to the highest-priority one and then clear the others without dealing with them?